ui/core/src/api/MarketService/index.ts

```diff
@@ -1,80 +1,67 @@
-import { RWN } from "../../constants";
-import { Asset, AssetAmount, Pool } from "../../entities";
-import { Fraction } from "../../entities/fraction/Fraction";
+import ReconnectingWebSocket from "reconnecting-websocket";
+import { Asset, Pool } from "../../entities";
 import { SifUnSignedClient } from "../utils/SifClient";
-import { RawPool } from "../utils/x/clp";
+import { toPool } from "../utils/toPool";
 
 export type MarketServiceContext = {
   loadAssets: () => Promise<Asset[]>;
   sifApiUrl: string;
   nativeAsset: Asset;
 };
 
-function toAssetSymbol(assetOrString: Asset | string) {
-  return typeof assetOrString === "string"
-    ? assetOrString
-    : assetOrString.symbol;
-}
-
-function makeQuerablePromise<T>(promise: Promise<T>) {
-  let isResolved = false;
-
-  promise.then(() => {
-    isResolved = true;
-  });
-
-  return {
-    isResolved() {
-      return isResolved;
-    },
-  };
-}
-
-function processPool(poolData: RawPool) {
-  const externalAssetTicker = poolData.external_asset.ticker;
-
-  return Pool(
-    AssetAmount(RWN, poolData.native_asset_balance),
-    AssetAmount(
-      Asset.get(externalAssetTicker),
-      poolData.external_asset_balance
-    ),
-    new Fraction(poolData.pool_units)
-  );
-}
+type PoolHandlerFn = (pools: Pool[]) => void;
 
 export default function createMarketService({
   loadAssets,
   sifApiUrl,
 }: MarketServiceContext) {
+  let ws: ReconnectingWebSocket;
   const sifClient = new SifUnSignedClient(sifApiUrl);
-  const poolMap = new Map<string, Pool>();
+
+  let poolHandler: PoolHandlerFn = () => {};
+
+  async function setupPoolWatcher() {
+    await new Promise((res, rej) => {
+      ws = new ReconnectingWebSocket("ws://localhost:26657/websocket");
+      ws.onopen = () => {
+        ws.send(
+          JSON.stringify({
+            jsonrpc: "2.0",
+            method: "subscribe",
+            id: "1",
+            params: {
+              query: `tm.event='Tx'`,
+            },
+          })
+        );
+        // This assumes every transaction means an update to pool values
+        // Subscribing to all pool addresses could mean having a tone of
+        // open connections to our node because there is no "OR" query
+        // syntax so have chosen to go with debouncing getPools for now.
+        ws.onmessage = async (...argoids) => {
```

Comment:
Makes sense. So what you're saying is eventually we'll want some logic to parse between messages?

Reply:
Well ideally we would subscribe to specific pools through a single connection so we only get notified when those specific pools are updated. Cosmos query language doesn’t allow for OR queries for some reason so the only way to do this is to subscribe over multiple channels. So one per query. If we have 50 coins that’s 50 connections per browser.